src/transformers/models/megatron_gpt2/convert_megatron_gpt2_checkpoint.py

```diff
@@ -0,0 +1,238 @@
+            output_state_dict[layer_name + "." + ln_name + "." + weight_or_bias] = val
+
+        # Transpose the QKV matrix.
+        elif op_name == "attention.query_key_value" and weight_or_bias == "weight":
+
+            # Insert a tensor of 1x1xDxD bias.
+            zeros = torch.ones(1, 1, hidden_size, hidden_size)
+            output_state_dict[layer_name + ".attn.bias"] = zeros
+
+            # Insert a "dummy" tensor for masked_bias.
+            masked_bias = torch.tensor(-1e4)
+            output_state_dict[layer_name + ".attn.masked_bias"] = masked_bias
+
+            # Megatron stores (3*D) x D but transformers-GPT2 expects D x 3*D.
+            out_val = val.transpose(0, 1)
+            # Store.
+            output_state_dict[layer_name + ".attn.c_attn.weight"] = out_val
+
+        # Transpose the bias.
+        elif op_name == "attention.query_key_value" and weight_or_bias == "bias":
+
+            # Store. No change of shape.
+            output_state_dict[layer_name + ".attn.c_attn.bias"] = val
+
+        # Transpose the weights.
+        elif weight_or_bias == "weight":
+
+            out_name = megatron_to_transformers[op_name]
+            output_state_dict[layer_name + out_name + "weight"] = val.transpose(0, 1)
+
+        # Copy the bias.
+        elif weight_or_bias == "bias":
+
+            out_name = megatron_to_transformers[op_name]
+            output_state_dict[layer_name + out_name + "bias"] = val
+
+    # The final layernorm.
+    output_state_dict["transformer.ln_f.weight"] = transformer["final_layernorm.weight"]
+    output_state_dict["transformer.ln_f.bias"] = transformer["final_layernorm.bias"]
+
+    # For LM head, transformers' wants the matrix to weight embeddings.
+    output_state_dict["lm_head.weight"] = word_embeddings
+
+    # The config.
+    output_config = {
+        "activation_function": "gelu_new",
+        "architectures": ["GPT2LMHeadModel"],
+        "attn_pdrop": 0.1,
+        "bos_token_id": 50256,
+        "embd_pdrop": 0.1,
+        "eos_token_id": 50256,
+        "initializer_range": 0.02,
+        "layer_norm_epsilon": 1e-05,
+        "model_type": "gpt2",
+        "n_ctx": 1024,
+        "n_embd": 1024,
+        "n_head": 16,
+        "n_layer": 24,
+        "n_positions": 1024,
+        "resid_pdrop": 0.1,
+        "summary_activation": None,
+        "summary_first_dropout": 0.1,
+        "summary_proj_to_labels": True,
+        "summary_type": "cls_index",
+        "summary_use_proj": True,
+        "vocab_size": 50257,
+    }
+
+    # It should be done!
+    return output_state_dict, output_config
+
+
+####################################################################################################
+
+
+def main():
+    # Create the argument parser.
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--print-checkpoint-structure", action="store_true")
+    parser.add_argument("path_to_checkpoint", type=str, help="Path to the ZIP file containing the checkpoint")
```

Comment:
Should this have the `--` prefix? 
```suggestion
    parser.add_argument("--path_to_checkpoint", type=str, help="Path to the ZIP file containing the checkpoint")
```

Reply:
[Copied from above ;)] If we add the -- it changes the command invocation from python3 convert.py <path_to_file> to python3 convert.py --path_to_checkpoint <path_to_file>. That's extra typing. Also, the <path_to_file> argument is mandatory so the positional argument is more common in that context. I would vote against adding the -- (but I can do it if you think it's needed/more coherent with the rest of Transformers).